pydoctor/astbuilder.py

```diff
@@ -807,24 +807,29 @@ def _handleFunctionDef(self,
         is_property = False
         is_classmethod = False
         is_staticmethod = False
-        if isinstance(parent, model.Class) and node.decorator_list:
+        is_overload_func = False
+        if node.decorator_list:
             for d in node.decorator_list:
                 if isinstance(d, ast.Call):
                     deco_name = node2dottedname(d.func)
                 else:
                     deco_name = node2dottedname(d)
                 if deco_name is None:
                     continue
-                if deco_name[-1].endswith('property') or deco_name[-1].endswith('Property'):
-                    is_property = True
-                elif deco_name == ['classmethod']:
-                    is_classmethod = True
-                elif deco_name == ['staticmethod']:
-                    is_staticmethod = True
-                elif len(deco_name) >= 2 and deco_name[-1] in ('setter', 'deleter'):
-                    # Rename the setter/deleter, so it doesn't replace
-                    # the property object.
-                    func_name = '.'.join(deco_name[-2:])
+                if isinstance(parent, model.Class):
+                    if deco_name[-1].endswith('property') or deco_name[-1].endswith('Property'):
+                        is_property = True
+                    elif deco_name == ['classmethod']:
+                        is_classmethod = True
+                    elif deco_name == ['staticmethod']:
+                        is_staticmethod = True
+                    elif len(deco_name) >= 2 and deco_name[-1] in ('setter', 'deleter'):
+                        # Rename the setter/deleter, so it doesn't replace
+                        # the property object.
+                        func_name = '.'.join(deco_name[-2:])
+                if deco_name[-1] == 'overload':
```

Comment:
```suggestion
                full_deco_name = parent.expandName('.'.join(deco_name))
                if full_deco_name == 'typing.overload':
```

This will check whether we use `typing.overload` and not just `something.overload`.

Reply:
I wasn't aware of the performance cost of qualifying this and I noticed we don't qualify on any other decorator name changes so I just assumed we were afraid of it. I will change to qualify the name before checking.

Reply:
We are not qualifying other decorators because they are either supposed to be part of the builtins (in which case the expanded name if the same) or we use a naming convention (for properties). 

> performance cost of qualifying this

Probably this method could be optimized, but it only looks in the objects in the system, it's not like it will parse additional modules to resolve a name. So it's ok to use it, the only thing to keep in mind is that at the time of AST building all objects are not in the system yet, so resolveName is not 100% reliable. But to resolve full import names it's definitely enough.

Reply:
Fixed